src/main/java/com/heroiclabs/nakama/WebSocketClient.java

```diff
@@ -74,7 +74,7 @@ public JsonElement serialize(final byte[] src, final Type typeOfSrc, final JsonS
     private final OkHttpClient client;
     private final Map<String, SettableFuture<?>> collationIds;
     private WebSocket socket;
-    private final ExecutorService listenerThreadPoolExec = Executors.newCachedThreadPool();
+    private final ExecutorService listenerThreadPoolExec = Executors.newSingleThreadExecutor();
```

Comment:
I think we need to provide an option in the Connect() to pass in a custom ExecutorService, and if not then use SingleThreadExecutor.

Reply:
@mofirouz I was wondering why since it would allow users to break the assumed ordering guarantees of the web socket wrapper.

Reply:
Because in some cases they'd like to use their own ExecutorService strategy (like only match data being single threaded, everything else multithreaded). This would allow for the flexibility and is a power user feature :)

Reply:
Okay, I can see the case for using multiple threads in cases where the message callbacks don't need to be sequenced. Since there isn't a way to currently map a certain threading model to a certain type of web socket message, would we need to implement that as well? Unless they were separate sockets, but then I'm not sure how the server would know which one to send certain messages through.